src/credentials/providers/environmentCredentialsProviderFactory.ts

```diff
@@ -0,0 +1,53 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { getLogger } from '../../shared/logger';
+import { BaseCredentialsProviderFactory } from "./credentialsProviderFactory";
+import { Ec2CredentialsProvider } from "./ec2CredentialsProvider";
+import { EnvironmentCredentialsProvider } from "./environmentCredentialsProvider";
+import { EnvVarsCredentialsProvider } from './envVarsCredentialsProvider';
+
+export class EnvironmentCredentialsProviderFactory extends BaseCredentialsProviderFactory<EnvironmentCredentialsProvider> {
```

Comment:
Naming nit: for consistency I suggest naming this `EnvCredentialsProviderFactory` or `EnvVars...`. Spelling out "environment" is pedantic because "env" is exceedingly common in computing with low risk of ambiguity.

(Out of scope for this PR) Now that we have implemented another "credentials factory", does it appear to do anything useful? Why couldn't we do all of this in the "manager"?